src/components/manage/History/History.jsx

```diff
@@ -116,22 +116,46 @@ class History extends Component {
     this.props.revertHistory(getBaseUrl(this.props.pathname), value);
   }
 
-  /**
-   * Render method.
-   * @method render
-   * @returns {string} Markup for the component.
-   */
-  render() {
+  processHistoryEntries = () => {
+    // Getting the history entries from the props
+    // No clue why the reverse(concat()) is necessary
     const entries = reverse(concat(this.props.entries));
     let title = entries.length > 0 ? entries[0].state_title : '';
     for (let x = 1; x < entries.length; x += 1) {
       entries[x].prev_state_title = title;
       title = entries[x].state_title || title;
     }
+    // We reverse them again
     reverse(entries);
+
+    // We identify the latest 'versioning' entry and mark the others with
+    // show the revert button property
+    let mark = false;
+    return entries.map((item) => {
+      if (item.type === 'versioning' && !mark) {
+        mark = true;
+        return item;
+      }
+
+      if (item.type === 'versioning' && mark) {
+        return { ...item, show_revert_button: true };
+      }
+
+      return item;
+    });
+  };
```

Comment:
You could use lodash.find to add `is_current: true` to the first item that has `type: 'versioning'` instead of adding `show_revert_true` to all the others. And then check for `is_current` below.

But, this is good enough.

Reply:
@davisagli refactored using `find`, I thought it would return all the matching elements. Thanks for the suggestion!